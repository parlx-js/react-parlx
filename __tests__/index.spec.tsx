import React from 'react';
import { render } from '@testing-library/react';

import ReactParlx from '../src';

describe('ReactParlx', () => {
  it('should render component with default props', () => {
    const { container } = render(<ReactParlx />);

    expect(container).toMatchSnapshot();
  });

  it('should render component with custom props', () => {
    const { container } = render(
      <ReactParlx
        className="parlx"
        style={{ border: '1px solid #f00' }}
        overlay
        overlayProps={{
          className: 'parlx-overlay',
        }}
      >
        <p>Hello World!</p>
      </ReactParlx>
    );

    expect(container).toMatchSnapshot();
  });
});
