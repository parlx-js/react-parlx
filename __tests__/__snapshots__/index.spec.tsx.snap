// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactParlx should render component 1`] = `
<div>
  <div
    class="parallax"
    style="height: 400px;"
  >
    <div
      class="parlx-children"
      style="height: 460.79999999999995px; width: 614.4px; transform: translate(0px, 0px); object-fit: cover;"
    />
  </div>
</div>
`;

exports[`ReactParlx should render component 2`] = `
<div>
  <div
    class="parlx"
    style="border: 1px solid #f00; height: 400px;"
  >
    <div
      class="parlx-overlay"
    />
    <p>
      Hello World!
    </p>
    <div
      class="parlx-children"
      style="height: 460.79999999999995px; width: 614.4px; transform: translate(0px, 0px); object-fit: cover;"
    />
  </div>
</div>
`;

exports[`ReactParlx should render component with custom props 1`] = `
<div>
  <div
    class="parlx"
    style="border: 1px solid #f00; height: 400px;"
  >
    <div
      class="parlx-overlay"
    />
    <p>
      Hello World!
    </p>
    <div
      class="parlx-children"
      style="height: 460.79999999999995px; width: 614.4px; transform: translate(0px, 0px); object-fit: cover;"
    />
  </div>
</div>
`;

exports[`ReactParlx should render component with default props 1`] = `
<div>
  <div
    class="parallax"
    style="height: 400px;"
  >
    <div
      class="parlx-children"
      style="height: 460.79999999999995px; width: 614.4px; transform: translate(0px, 0px); object-fit: cover;"
    />
  </div>
</div>
`;
